rust/hyperlane-core/src/types/announcement.rs

```diff
@@ -0,0 +1,87 @@
+use ethers::{
+    prelude::{Address, Signature},
+    utils::hash_message,
+};
+use ethers_signers::Signer;
+use serde::{Deserialize, Serialize};
+use sha3::{Digest, Keccak256};
+
+use crate::{utils::domain_hash, HyperlaneProtocolError, SignerExt, H256};
+
+/// An Hyperlane checkpoint
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct Announcement {
+    /// The mailbox address
+    pub mailbox_address: H256,
+    /// The mailbox chain
+    pub mailbox_domain: u32,
+    /// The checkpointed root
```

Comment:
nit: seems incorrect, also comment on line 11

Reply:
"seems" is generous lol